### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by ErikStoupignan**
# Add devise gem 💎 

This branch was used to apply and configure the "devise" gem in our project.

## Project requirements
- Remove current_user method in ApplicationController, devise will provide us with one. ✅ 
- Install and setup devise. ✅ 
  - Can register a new user. ✅ 
  - User logs in with a combination of email and password. ✅ 
  - Hashed password should be stored in the database. ✅ 
  - Ask for confirmation of email. ✅ 
  - Can reset password. ✅ 
- Modify the views in devise for registration and login to match the wireframes from the Sneak Peek and your styling. ✅ 

## Instructions:
- Using the url: http://localhost:3000/ you will be redirected to the link: http://localhost:3000/users/sign_in. 
- In this window you will find a link with the name [sign up](http://localhost:3000/users/sign_up) that will redirect you to the page to give your registration information in the system.
- Once you are in the "Sign up" window, please enter all your data following the instructions on the screen.
- If everything goes well, you will receive a notification in your visual studio console with a link so you can confirm.

### Inserting data:
image

### Receiving notification in the Visual Studio console:
image

### Or if you don't see the link, you can click here http://localhost:3000/users/confirmation/new to ask for another notification on your visual studio.
image

### And ready! The devise gem would already be working perfectly 😄 
image

### Don't forget to log out and refresh your browser when you're done! 
image

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
+</div>
```
Comment (codecaiine) [inline app/views/devise/registrations/new.html.erb]: - [x] Good job so far I am able to register and log in :clap: But there are a few things about your design. Please make sure that your `registration `and `login` views match the wireframes and their styling as expected for this project. As a developer it is very important to pay attention to details :smile: 

#### Login example
blog_login

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @ErikStoupignan @AmaliaGM , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Devise well setup :heavy_check_mark: 
- Registration and login work as expected :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 
- Good use of gitflow :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — COMMENTED** [review]
Hi @AmaliaGM, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that you should submit your PR link to your project this is your partner project but you should also both implement it in your repo, since the requirements is to make it in all the repostires together so kindly also implement it in your send then send your PR ✨💯🔥🔥

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by andersonlebon — APPROVED** [review]
### Hello @ErikStoupignan 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [x] [OPTIONAL] Kindly make sure you removed the template logo. Optionally, you can replace it with your own logo if you have one.
Screenshot from 2022-12-07 03-36-21

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

**Review by Reem-lab — APPROVED** [review]
# Hi @ErikStoupignan, 👋🏻👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🚀🔥🎉🎉 


## Highlights ✨✨
- No Linters error ✅
- Followed Gitflow ✅
- Great working on the functionality and the design🔥🚀🎉
- Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**